src/data_retrieval/__main__.py

```diff
@@ -1,15 +1,16 @@
 import argparse
 import os
-from typing import List
+from typing import Optional
 
-import pandas as pd
+from azure_blob_utils import load_env_vars, upload_stream
 from cds.ecmwf import download_ecmwf_cds
 from cds.era5 import download_era5_cds
 from cds.mars import download_ecmwf_mars, get_country_bbox_df
 from util import get_logger, setup_output_path
 
-logger = get_logger(__name__)
+sas_token, container_name, storage_account = load_env_vars()
```

Comment:
The way you implemented `load_env_vars()` is very good. But you should not call it form here.
If user wants to keep using the tool locally because they do not have Azure credentials, they will still have to set the env vars to something even when they will never use it. Otherwise the tool crash with error.

That is bad design and technically a bug. Please call this function only when `--upload` is used. Then it will make sense to throw the error if they want to upload the file to Azure and did not provide the env vars.